pandas/_libs/src/parser/io.c

```diff
@@ -66,6 +66,9 @@ void *buffer_rd_bytes(void *source, size_t nbytes, size_t *bytes_read,
     args = Py_BuildValue("(i)", nbytes);
 
     func = PyObject_GetAttrString(src->obj, "read");
+    if (func == NULL) {
+        return NULL;
```

Comment:
It looks like you also need to `Py_DECREF(args)` - looks to be Py_XDECREF a few lines down but won't happen if you return here.

As a side note - I think we also too liberally use `Py_XDECREF`. But that's another issue for another day